app/controllers/devise_recognisable/sessions_controller.rb

```diff
@@ -42,6 +46,21 @@ def check_for_authentication_token
       self.resource = resource_class.find token['id']
 
       sign_in self.resource
+      DeviseRecognisable::RecognisableSession.create!(
```

Comment:
Does this create two of these because it's done in the `after_action` too?

Reply:
@jalada I don't think so. The after action only runs after `:create` as this method, `:perform_ip_check` runs before `:new`.

Reply:
Lol Github diff collapse.